src/smartpeak/source/core/RawDataProcessor.cpp

```diff
@@ -1844,4 +1844,206 @@ namespace SmartPeak
+        }
+      }
+      
+      isotopelabelingmdvs.calculateMDVs(rawDataHandler_IO.getFeatureMap(), normalized_featureMap, mass_intensity_type, feature_name);
+      rawDataHandler_IO.setFeatureMap(normalized_featureMap);
+    }
+    catch (const std::exception& e) {
+      LOGE << e.what();
+    }
+
+    LOGD << "END calculateMDVs";
+  }
+
+  void IsotopicCorrections::process( // TODO:captitalise
+    RawDataHandler& rawDataHandler_IO,
+    const std::map<std::string, std::vector<std::map<std::string, std::string>>>& params_I,
+    const Filenames& filenames
+  ) const
+  {
+    LOGD << "START IsotopicCorrections";
+
+    if (params_I.count("IsotopicCorrections") && params_I.at("IsotopicCorrections").empty()) {
+      LOGE << "No parameters passed to IsotopicCorrections.";
+      LOGD << "END IsotopicCorrections";
+      return;
+    }
+
+    OpenMS::IsotopeLabelingMDVs isotopelabelingmdvs;
+    OpenMS::Param parameters = isotopelabelingmdvs.getParameters();
+    isotopelabelingmdvs.setParameters(parameters);
+
+    try {
+      OpenMS::FeatureMap corrected_featureMap;
+      std::vector<std::map<std::string, std::string>> IsotopicCorrections_params = params_I.find("IsotopicCorrections")->second;
+      
+      std::string correction_matrix_agent;
+      for(auto& param : IsotopicCorrections_params)
+      {
+        if (param.find("name")->second == "correction_matrix_agent")
+        {
+          correction_matrix_agent = param.find("value")->second;
+        }
+      }
+      
+      isotopelabelingmdvs.isotopicCorrections(rawDataHandler_IO.getFeatureMap(), corrected_featureMap, {}, correction_matrix_agent);
+      rawDataHandler_IO.setFeatureMap(corrected_featureMap);
+    }
+    catch (const std::exception& e) {
+      LOGE << e.what();
+    }
+
+    LOGD << "END IsotopicCorrections";
+  }
+
+  void CalculateIsotopicPurities::process(
+    RawDataHandler& rawDataHandler_IO,
+    const std::map<std::string, std::vector<std::map<std::string, std::string>>>& params_I,
+    const Filenames& filenames
+  ) const
+  {
+    LOGD << "START calculateIsotopicPurities";
+
+    if (params_I.count("CalculateIsotopicPurities") && params_I.at("CalculateIsotopicPurities").empty()) {
+      LOGE << "No parameters passed to CalculateIsotopicPurities.";
+      LOGD << "END CalculateIsotopicPurities";
+      return;
+    }
+
+    OpenMS::IsotopeLabelingMDVs isotopelabelingmdvs;
+    OpenMS::Param parameters = isotopelabelingmdvs.getParameters();
+    isotopelabelingmdvs.setParameters(parameters);
+
+    try {
+      OpenMS::FeatureMap normalized_featureMap;
+      std::vector<std::map<std::string, std::string>> CalculateIsotopicPurities_params = params_I.find("CalculateIsotopicPurities")->second;
+      
+      std::string experiment_data_s, isotopic_purity_name;
+      std::vector<double> experiment_data;
+      for(auto& param : CalculateIsotopicPurities_params)
+      {
+        if (!param.find("comment_")->second.empty())
```

Comment:
What is the reason for using the `comment_` column?   It is actually not supposed to be used for the data processing, but only for the user when directly annotating the parameters file.

Reply:
The `calculateMDVAccuracies(..)` takes in a 2 vectors of doubles for the measured and theoretical data, could we embed these data in other `csv` files ?